Desktop_Interface/functiongencontrol.cpp

```diff
@@ -86,65 +83,103 @@ void functionGenControl::waveformName(QString newName, ChannelData& channel, int
     QByteArray remainingData = fptr.readAll();
     char *dataString = remainingData.data();
 
-    free(channel.samples);
-    channel.samples = (unsigned char *) malloc(channel.length);
+	channel.samples.resize(channel.length);
```

Comment:
I don't think it makes sense to store the length as a member variable in the channels struct, since we now have a `std::vector` with a `size()`method